FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+                    break;
+                }
+
+                /* Increment the number of responses found on the queue. */
+                xNumResponses += 1;
+                /* Reset the wait counter every time a response is received. */
+                ulWaitCounter = 0;
+            }
+
+            /* Break if we have received all expected responses. */
+            if( xNumResponses >= xResponseCount )
+            {
+                break;
+            }
+
+            /* Break if we have been stuck waiting for a response for too long.
+             * The total wait here will be the (notification check delay + queue
+             * check delay), multiplied by `httpexampleMAX_WAIT_ITERATIONS`. For
+             * example, with a 1000 ms delay for both checks, and a maximum
+             * iteration of 5, this function will wait 10 seconds after
+             * receiving the last response before exiting the loop. */
+            if( ++ulWaitCounter > httpexampleMAX_WAIT_ITERATIONS )
+            {
+                LogError( ( "Response receive loop exceeded maximum wait time." ) );
+                break;
+            }
+            else if( prvCheckNotification( &ulNotification, httpexampleHTTP_FAILURE, pdTRUE ) != pdFALSE )
+            {
+                LogError( ( "Response task: Received error notification from the main HTTP task. Exiting task." ) );
+                break;
+            }
+        }
+    }
+
+    /* Clear this task's notifications. */
+    xTaskNotifyStateClear( NULL );
+    ulNotification = ulTaskNotifyValueClear( NULL, ~( 0U ) );
+
+    /* Notify the other tasks of completion. */
+    xTaskNotify( xMainTask, httpexampleRESPONSE_TASK_COMPLETION, eSetBits );
+    xTaskNotify( xResponseTask, httpexampleRESPONSE_TASK_COMPLETION, eSetBits );
+
+    /* Delete this task. */
+    LogInfo( ( "Deleting response task." ) );
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvCheckNotification( uint32_t * pulNotification,
+                                        uint32_t ulExpectedBits,
+                                        BaseType_t xClearBits )
+{
+    BaseType_t xStatus = pdTRUE;
+
+    configASSERT( pulNotification != NULL );
+
+    xTaskNotifyWait( 0,
+                     ( xClearBits == pdTRUE ) ? ulExpectedBits : 0,
+                     pulNotification,
+                     httpexampleDEMO_TICKS_TO_WAIT );
+
+    xStatus = ( ( *pulNotification & ulExpectedBits ) == ulExpectedBits ) ? pdTRUE : pdFALSE;
+
+    return xStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvDownloadLoop( void )
+{
+    /* The transport layer interface used by the HTTP client library. */
+    TransportInterface_t xTransportInterface;
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+    BaseType_t xStatus = pdPASS;
+    uint32_t ulNotification = 0U;
```

Comment:
```suggestion
    uint32_t ulNotification = 0UL;
```

Reply:
Thanks, have fixed these in the "Addressing Feedback" commit.